v2/service/module/watcher/watcher.go

```diff
@@ -0,0 +1,95 @@
+// Package watcher provides file update notification functionality as a module.
+package watcher
+
+import (
+	"fmt"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+var (
+	ErrMissingWithFilename = fmt.Errorf("watcher.Watcher missing WithFilename option")
+	ErrMissingWithFunc     = fmt.Errorf("watcher.Watcher missing WithFunc option")
+)
+
+type Watcher struct {
+	w        *fsnotify.Watcher
+	stopped  chan struct{}
+	filename *string
```

Comment:
This could be just `string` instead of `*string` since empty string is also invalid value.